relayer/src/ethereum_checkpoints/tests/mod.rs

```diff
@@ -0,0 +1,440 @@
+use super::utils::{self, slots_batch, FinalityUpdateResponse};
+use checkpoint_light_client::WASM_BINARY;
+use checkpoint_light_client_io::{
+    ethereum_common::{
+        base_types::BytesFixed, network::Network, utils as eth_utils, SLOTS_PER_EPOCH,
+    },
+    replay_back, sync_update,
+    tree_hash::TreeHash,
+    Handle, HandleResult, Init, G2,
+};
+use gclient::{EventListener, EventProcessor, GearApi, Result};
+use parity_scale_codec::{Decode, Encode};
+use reqwest::Client;
+use std::env;
+use tokio::time::{self, Duration};
+
+const RPC_URL: &str = "http://127.0.0.1:5052";
+
+const FINALITY_UPDATE_5_254_112: &[u8; 4_940] =
+    include_bytes!("./sepolia-finality-update-5_254_112.json");
+const FINALITY_UPDATE_5_263_072: &[u8; 4_941] =
+    include_bytes!("./sepolia-finality-update-5_263_072.json");
+
+async fn common_upload_program(
+    client: &GearApi,
+    code: Vec<u8>,
+    payload: impl Encode,
+) -> Result<([u8; 32], [u8; 32])> {
+    let encoded_payload = payload.encode();
+    let gas_limit = client
+        .calculate_upload_gas(None, code.clone(), encoded_payload, 0, true)
+        .await?
+        .min_limit;
+    println!("init gas {gas_limit:?}");
+    let (message_id, program_id, _) = client
+        .upload_program(
+            code,
+            gclient::now_micros().to_le_bytes(),
+            payload,
+            gas_limit,
+            0,
+        )
+        .await?;
+
+    Ok((message_id.into(), program_id.into()))
+}
+
+async fn upload_program(
+    client: &GearApi,
+    listener: &mut EventListener,
+    payload: impl Encode,
+) -> Result<[u8; 32]> {
+    let (message_id, program_id) =
+        common_upload_program(client, WASM_BINARY.to_vec(), payload).await?;
+
+    assert!(listener
+        .message_processed(message_id.into())
+        .await?
+        .succeed());
+
+    Ok(program_id)
+}
+
+#[tokio::test]
```

Comment:
Please add `ignore` for these tests as I see that they'll fail without node running on `localhost`